library/math/conversions.lisp

```diff
@@ -182,5 +182,32 @@ cannot be represented in :TO. These fall into a few categories:
                  (Err "Integer to Double-Float conversion out-of-range")
                  (Ok y)))))))
 
+(cl:defmacro integer-tryinto-float (integer integer-name lisp-float float float-name pow)
```

Comment:
Have the macro generate a `define-instance` without a  `coalton-toplevel`. Then call the macro inside a toplevel like:
```
(coalton-toplevel
  (integer-tryinto-float ...)
  ...)
```
This allows the Coalton compiler to gives better error messages.